code/mob/living/silicon/ai.dm

```diff
@@ -684,7 +685,13 @@ var/list/ai_emotions = list("Happy" = "ai_happy",\
 /mob/living/silicon/ai/triggerAlarm(var/class, area/A, var/O, var/alarmsource)
 	if (isdead(src))
 		return 1
-	var/list/L = src.alarms[class]
+	var/indexclass = class
+	for(var/loopclass in list("Fire", "Low Pressure", "Flammable Atmosphere", "Flood", "Manual Trip"))
+		if(class == loopclass)
+			indexclass = "Fire"
+			break
+
+	var/list/L = src.alarms[indexclass]
```

Comment:
This block of code isn't doing anything?

Reply:
It is, it's just very stupid. This line of code is making it so all unique alarms are classified as fire alarms for the purposes of the list index. IIRC this index also tracks motion-detecting cameras.

Reply:
shitcode